==== Unchecked return value from external call. ====

SWC ID: 104

Severity: Medium

Contract: owned

Function name: bug_unchk_send4()

PC address: 204

Estimated Gas Usage: 852 - 35040

The return value of a message call is not checked.

External calls return a boolean value. If the callee halts with an exception, 'false' is returned and execution continues in the caller. The caller should check whether an exception happened and react accordingly to avoid unexpected behavior. For example it is often desirable to wrap external calls in require() so the transaction is reverted if the call fails.

--------------------

In file: buggy_36.sol:25



msg.sender.send(1 ether)



--------------------

Initial State:



Account: [CREATOR], balance: 0x1, nonce:0, storage:{}

Account: [ATTACKER], balance: 0x0, nonce:0, storage:{}

Account: [SOMEGUY], balance: 0x0, nonce:0, storage:{}



Transaction Sequence:



Caller: [CREATOR], calldata: , value: 0x0

Caller: [CREATOR], function: transferOwnership(address), txdata: 0xf2fde38b0000000000000000000000000000000000000000000000000000000000000000, value: 0x0

Caller: [SOMEGUY], function: bug_unchk_send32(), txdata: 0x58d9161f, value: 0xde0b6b3a7640000



==== Unprotected Ether Withdrawal ====

SWC ID: 105

Severity: High

Contract: owned

Function name: bug_unchk_send4()

PC address: 204

Estimated Gas Usage: 852 - 35040

Any sender can withdraw Ether from the contract account.

Arbitrary senders other than the contract creator can profitably extract Ether from the contract account. Verify the business logic carefully and make sure that appropriate security controls are in place to prevent unexpected loss of funds.

--------------------

In file: buggy_36.sol:25



msg.sender.send(1 ether)



--------------------

Initial State:



Account: [CREATOR], balance: 0x2280000016a000400, nonce:0, storage:{}

Account: [ATTACKER], balance: 0x20020c00160006960, nonce:0, storage:{}

Account: [SOMEGUY], balance: 0x0, nonce:0, storage:{}



Transaction Sequence:



Caller: [CREATOR], calldata: , value: 0x0

Caller: [SOMEGUY], function: bug_unchk_send32(), txdata: 0x58d9161f, value: 0xde0b6b3a7640001

Caller: [ATTACKER], function: bug_unchk_send32(), txdata: 0x58d9161f, value: 0xde0b6b3a763ffff



==== External Call To User-Supplied Address ====

SWC ID: 107

Severity: Low

Contract: tokenRecipient

Function name: receiveApproval(address,uint256,address,bytes)

PC address: 446

Estimated Gas Usage: 3561 - 43087

A call to a user-supplied address is executed.

An external message call to an address specified by the caller is executed. Note that the callee account might contain arbitrary code and could re-enter any function within this contract. Reentering the contract in an intermediate state may lead to unexpected behaviour. Make sure that no state modifications are executed after this call and/or reentrancy guards are in place.

--------------------

In file: buggy_36.sol:42



t.transferFrom(_from, address(this), _value)



--------------------

Initial State:



Account: [CREATOR], balance: 0x20, nonce:0, storage:{}

Account: [ATTACKER], balance: 0x0, nonce:0, storage:{}

Account: [SOMEGUY], balance: 0x0, nonce:0, storage:{}



Transaction Sequence:



Caller: [CREATOR], calldata: , value: 0x0

Caller: [ATTACKER], function: receiveApproval(address,uint256,address,bytes), txdata: 0x8f4ffcb10000000000000000000000000000000000000000000000000000000000000000000000000000002f000000000000000000000000000000000000000000000000000000000000000000000000deadbeefdeadbeefdeadbeefdeadbeefdeadbeef0000000000000000000000000000000000000000000000000000000000000008, value: 0x0



==== Unchecked return value from external call. ====

SWC ID: 104

Severity: Medium

Contract: tokenRecipient

Function name: bug_unchk_send30()

PC address: 742

Estimated Gas Usage: 940 - 35128

The return value of a message call is not checked.

External calls return a boolean value. If the callee halts with an exception, 'false' is returned and execution continues in the caller. The caller should check whether an exception happened and react accordingly to avoid unexpected behavior. For example it is often desirable to wrap external calls in require() so the transaction is reverted if the call fails.

--------------------

In file: buggy_36.sol:46



msg.sender.send(1 ether)



--------------------

Initial State:



Account: [CREATOR], balance: 0x80, nonce:0, storage:{}

Account: [ATTACKER], balance: 0x8000000000004000, nonce:0, storage:{}

Account: [SOMEGUY], balance: 0x0, nonce:0, storage:{}



Transaction Sequence:



Caller: [CREATOR], calldata: , value: 0x0

Caller: [ATTACKER], function: bug_unchk_send30(), txdata: 0xdc52a049, value: 0xde0b6b3a7640000



==== Exception State ====

SWC ID: 110

Severity: Medium

Contract: MindsyncPlatform

Function name: proposals(uint256)

PC address: 2371

Estimated Gas Usage: 1074 - 1169

An assertion violation was triggered.

It is possible to trigger an assertion violation. Note that Solidity assert() statements should only be used to check invariants. Review the transaction trace generated for this issue and either make sure your program logic is correct, or use require() instead of assert() if your goal is to constrain user inputs or enforce preconditions. Remember to validate inputs from both callers (for instance, via passed arguments) and callees (for instance, via return values).

--------------------

In file: buggy_36.sol:167



Proposal[] public proposals



--------------------

Initial State:



Account: [CREATOR], balance: 0xc000007b987, nonce:0, storage:{}

Account: [ATTACKER], balance: 0x0, nonce:0, storage:{}

Account: [SOMEGUY], balance: 0x0, nonce:0, storage:{}



Transaction Sequence:



Caller: [CREATOR], calldata: , value: 0x0

Caller: [SOMEGUY], function: proposals(uint256), txdata: 0x013cf08b0000000000000000000000000000000000000000000000000000000000000000, value: 0x0



==== Unchecked return value from external call. ====

SWC ID: 104

Severity: Medium

Contract: MindsyncPlatform

Function name: bug_unchk_send1()

PC address: 2633

Estimated Gas Usage: 985 - 35173

The return value of a message call is not checked.

External calls return a boolean value. If the callee halts with an exception, 'false' is returned and execution continues in the caller. The caller should check whether an exception happened and react accordingly to avoid unexpected behavior. For example it is often desirable to wrap external calls in require() so the transaction is reverted if the call fails.

--------------------

In file: buggy_36.sol:479



msg.sender.send(1 ether)



--------------------

Initial State:



Account: [CREATOR], balance: 0xde8000000000002, nonce:0, storage:{}

Account: [ATTACKER], balance: 0xde0b6b3a7640000, nonce:0, storage:{}

Account: [SOMEGUY], balance: 0x0, nonce:0, storage:{}



Transaction Sequence:



Caller: [CREATOR], calldata: , value: 0x0

Caller: [ATTACKER], function: bug_unchk_send4(), txdata: 0x82caa706, value: 0xde0b6b3a7640000



==== Unprotected Ether Withdrawal ====

SWC ID: 105

Severity: High

Contract: MindsyncPlatform

Function name: bug_unchk_send1()

PC address: 2633

Estimated Gas Usage: 985 - 35173

Any sender can withdraw Ether from the contract account.

Arbitrary senders other than the contract creator can profitably extract Ether from the contract account. Verify the business logic carefully and make sure that appropriate security controls are in place to prevent unexpected loss of funds.

--------------------

In file: buggy_36.sol:479



msg.sender.send(1 ether)



--------------------

Initial State:



Account: [CREATOR], balance: 0x10, nonce:0, storage:{}

Account: [ATTACKER], balance: 0x20200373430100000, nonce:0, storage:{}

Account: [SOMEGUY], balance: 0x0, nonce:0, storage:{}



Transaction Sequence:



Caller: [CREATOR], calldata: , value: 0x1

Caller: [ATTACKER], function: bug_unchk_send18(), txdata: 0xd7e8c76e, value: 0xde0b6b3a763ffff





